bakery/src/scripts/disassemble_single.py

```diff
@@ -0,0 +1,193 @@
+            [role="doc-footnote"] {
+                background-color: #ffcccc;
+                border: 1px dashed #ff0000;
+            }
+            [role="doc-footnote"]:before { content: "FOOTNOTE " ; }
+
+            /* Show a permalink when hovering over a heading or paragraph */
+            *:not(:hover) > a.-dev-permalinker { display: none; }
+            * > a.-dev-permalinker {
+                margin-left: .1rem;
+                text-decoration: none;
+            }
+        '''
+
+        script.text = u'''//<![CDATA[
+            // SCRIPTS_FOR_DEVS
+            window.addEventListener('load', () => {
+                const pilcrow = '¶'
+
+                function addPermalink(parent, id) {
+                    const link = window.document.createElement('a')
+                    link.classList.add('-dev-permalinker')
+                    link.setAttribute('href', '#' + id)
+                    link.textContent = pilcrow
+                    parent.appendChild(link)
+                }
+
+                const paragraphs = Array.from(
+                    document.querySelectorAll('p[id]')
+                )
+                paragraphs.forEach(p => addPermalink(p, p.getAttribute('id')) )
+
+                const headings = Array.from(
+                    document.querySelectorAll(
+                        '*[id] > h1, *[id] > h2, *[id] > h3, ' +
+                        '*[id] > h4, *[id] > h5, *[id] > h6'
+                    )
+                )
+                headings.forEach(h => addPermalink(
+                    h, h.parentElement.getAttribute('id'))
+                )
+            })
+        // ]]>'''
+
+        head.append(style)
+        head.append(script)
+
+        with open(f"{out_dir / id_with_context}.xhtml", "wb") as out:
+            out.write(etree.tostring(root))
+
+        with open(
+            f"{out_dir / id_with_context}-metadata.json", "w"
+        ) as json_out:
+            # Incorporate metadata from disassemble step while setting defaults
+            # for cases like composite pages which may not have metadata from
+            # previous stages
+            json_metadata = {
+                "slug": slugs.get(doc.ident_hash),
+                "title": doc.metadata.get("title"),
+                "abstract": None,
+                "id": doc.id,
+                "revised": datetime.now().isoformat()
+            }
+
+            # Add / override metadata from baking if available
+            json_metadata.update(baked_metadata.get(doc.ident_hash, {}))
+
+            json.dump(
+                json_metadata,
+                json_out
+            )
+
+    with open(f"{out_dir}/{book_slug}.toc.xhtml", "wb") as out:
+        out.write(etree.tostring(toc, encoding="utf8", pretty_print=True))
+
+    with open(f"{out_dir}/{book_slug}.toc-metadata.json", "w") as toc_json:
+        json.dump(book_toc_metadata, toc_json)
+
+
+if __name__ == "__main__":
```

Comment:
Minor comment: This script looks pretty similar to `disassemble_book.py`. I'm not sure if there's going to be a need to maintain these in parallel for some non-trivial period, but if so (or it's unknown) maybe the shared stuff can be refactored in some manner? 🤷‍♂️  (if we think this is worth doing, it can be a separate issue / PR, just want to raise the thought)

Reply:
hmmm yeah... in fact, it's nearly the exact same, the only meaningful differences the inputs and outputs and preventing the using the `collection` part of the output basename. A diff:
```
$ diff ./scripts/disassemble_book.py ./scripts/disassemble_single.py
42,47c42,45
<     in_dir = Path(sys.argv[1]).resolve(strict=True)
<     out_dir = (in_dir / "disassembled").resolve(strict=True)
<     collection_uuid, collection_version = sys.argv[2:4]
<     baked_file = (in_dir / "collection.baked.xhtml").resolve(strict=True)
<     baked_metdata_file = \
<         (in_dir / "collection.baked-metadata.json").resolve(strict=True)
---
>     xhtml_file = Path(sys.argv[1]).resolve(strict=True)
>     metadata_file = Path(sys.argv[2]).resolve(strict=True)
>     book_slug = sys.argv[3]
>     out_dir = Path(sys.argv[4])
49c47
<     with open(baked_file, "rb") as file:
---
>     with open(xhtml_file, "rb") as file:
52d49
< 
55c52
<     with open(baked_metdata_file, "r") as baked_json:
---
>     with open(metadata_file, "r") as baked_json:
70c67
<         id_with_context = f'{collection_uuid}@{collection_version}:{doc.id}'
---
>         id_with_context = f'{binder.ident_hash}:{doc.id}'
188c185
<     with open(f"{out_dir}/collection.toc.xhtml", "wb") as out:
---
>     with open(f"{out_dir}/{book_slug}.toc.xhtml", "wb") as out:
191c188
<     with open(f"{out_dir}/collection.toc-metadata.json", "w") as toc_json:
---
>     with open(f"{out_dir}/{book_slug}.toc-metadata.json", "w") as toc_json:
```

Maybe there's a way to change disassemble_book to accept both tasks, kind of like we did with assemble-meta. I'll take a look.

Reply:
Okay, I did my best, but it did require a change in some regression test data. Would appreciate another set of eyes on that.